### PR TITLE
Cleanup Applications controller

### LINKED ISSUE #36: New application create, allow for mixed case users
perhaps make a select dropdown instead of field fill in

### LINKED ISSUE #53: Identify repeated view code to extract to partials

### PR DESCRIPTION
**Pull request by ETBassist**
- Create partial for showing a users details
- Move user details partial to shared dir and delete partials dir
- Create partial for showing a users review
- Create test to check that creating an application is not case sensitive
- Correct path expectation in case insensitive user name test
- Change method for finding user for an application to ignore differences in capitilization
- Extract user validity check into helper method
- Create test for adding pet to application helper method
- Update test to use id instead of full object
- Create helper method for adding a pet to an application
- Change to use application helper method to add pets to an application
- Create helper method on Pet to find by name
- Change controller to use Pet model method to find pets

### CODE REVIEW

File: app/models/pet.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ def self.count_of_pets
     count
   end
 
+  def self.with_names_like(name)
+    Pet.where('lower(name) like ?', "%#{name.downcase}%")
```
Comment (helloeduardo) [inline app/models/pet.rb]: Could even remove the `Pet` as it assumes self here 🥇

File: app/models/application.rb [hunk 2]
```diff
@@ -10,4 +10,9 @@ def has_pets?
   def has_pet?(pet)
     pets.exists?(pet.id)
   end
+
+  def add_pet(pet_id)
+    pet = Pet.find(pet_id)
+    pets << pet
```
Comment (helloeduardo) [inline app/models/application.rb]: Could one-liner this if we wanted to 🙌
```ruby
pets << Pet.find(pet_id)
```

**Review by helloeduardo — APPROVED** [review]
Left two comments on potential improvements! None are necessary however.

File: app/models/application.rb (hunk 2 above)
Comment (ETBassist) [reply app/models/application.rb]: Ooh good, lemme do that real quick

File: app/views/users/_show_review.html.erb [hunk 3]
```diff
@@ -1,4 +1,8 @@
-<%= review.title %>
-<%= review.rating %>
-<%= review.content %>
-<%= image_tag review.image if review.image %>
+<div class="review">
```
Comment (katemorris) [inline app/views/users/_show_review.html.erb:1]: Thank you!

**Review by katemorris — APPROVED** [review]

**Comment by katemorris** [comment]
Wooo, now the difficult part for me.

**Pull request merged**